activation/nipost.go

```diff
@@ -240,20 +210,27 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.NIPos
 		}
 
 		events.EmitPoetWaitProof(challenge.PublishEpoch, challenge.TargetEpoch(), time.Until(poetRoundEnd))
-		poetProofRef, membership, err := nb.getBestProof(ctx, nb.state.Challenge, challenge.PublishEpoch)
+		poetProofRef, membership, err := nb.getBestProof(ctx, challenge.Hash(), challenge.PublishEpoch)
 		if err != nil {
 			return nil, &PoetSvcUnstableError{msg: "getBestProof failed", source: err}
 		}
 		if poetProofRef == types.EmptyPoetProofRef {
 			return nil, &PoetSvcUnstableError{source: ErrPoetProofNotReceived}
 		}
-		nb.state.PoetProofRef = poetProofRef
-		nb.state.NIPost.Membership = *membership
-		nb.persistState()
+		if err := nipost.UpdatePoetProofRef(nb.localDB, nb.signer.NodeID(), poetProofRef); err != nil {
+			// TODO: panic instead?
```

Comment:
`nb.persistState()` did also only log on warning that it couldn't persist but would continue.